src/components/TimePicker/index.js

```diff
@@ -14,170 +14,148 @@ import StyledModal from './styled/modal';
-        if (prevValue !== value) {
-            this.updateValue();
-        }
-    }
-
-    getTriggerInputValue() {
-        const { value } = this.state;
-        const { placeholder, hour24 } = this.props;
+const TimePicker = props => {
+    const {
+        placeholder,
+        label,
+        required,
+        style,
+        className,
+        hideLabel,
+        name,
+        bottomHelpText,
+        isCentered,
+        error,
+        readOnly,
+        disabled,
+        tabIndex,
+        id,
+        cancelLabel,
+        okLabel,
+        onChange,
+        hour24,
+    } = props;
+    const [isOpen, setIsOpen] = useState(false);
+    const [value, setValue] = useState(hour24 ? props.value : get12HourTime(props.value));
+    const inputRef = React.createRef();
+    const timeSelectRef = React.createRef();
+    const prevValue = useRef(value);
+
+    const getTriggerInputValue = () => {
         return getInputValue(value, placeholder, hour24);
-    }
+    };
 
-    setFocusToHourInput() {
-        this.timeSelectRef.current.focusHourInput();
-    }
+    const setFocusToHourInput = () => {
+        timeSelectRef.current.focusHourInput();
+    };
 
-    updateValue() {
-        const { value, hour24 } = this.props;
-        this.setState({
-            value: hour24 ? value : get12HourTime(value),
-        });
-    }
+    const updateValue = () => {
+        setValue(hour24 ? props.value : get12HourTime(props.value));
+    };
 
-    handleKeyDown(event) {
+    const handleKeyDown = event => {
         const { keyCode } = event;
-        const { readOnly } = this.props;
         const shouldOpenModal = (keyCode === ENTER_KEY || keyCode === SPACE_KEY) && !readOnly;
         if (shouldOpenModal) {
-            this.setState({ isOpen: true });
+            setIsOpen(true);
         }
-    }
+    };
 
-    handleClick(event) {
-        const { onClick, readOnly } = this.props;
+    const handleClick = event => {
+        const { onClick } = props;
         if (!readOnly) {
-            this.setState({ isOpen: true });
+            setIsOpen(true);
             onClick(event);
         }
-    }
+    };
 
-    handleBlur() {
-        const { onBlur, value } = this.props;
+    const handleBlur = () => {
+        const { onBlur } = props;
         onBlur(value);
-    }
+    };
 
-    handleFocus() {
-        const { onFocus, value } = this.props;
+    const handleFocus = () => {
+        const { onFocus } = props;
         onFocus(value);
-    }
-
-    closeModal() {
-        this.setState({ isOpen: false });
-    }
+    };
+
+    const closeModal = () => {
+        setIsOpen(false);
+    };
+    useEffect(() => {
+        if (prevValue !== props.value) {
+            updateValue();
+        }
+    });
 
     /**
      * Sets focus on the element.
      * @public
      */
-    focus() {
-        this.inputRef.current.focus();
-    }
+    const focus = () => {
+        inputRef.current.focus();
+    };
```

Comment:
you must use useImperativeHandle () to export these functions.

You can be guided by PhoneInput